### DISCUSSION TITLE
Update framework for PHP 8+?

### DISCUSSION THREAD

**Discussion by Messhias** [laravel/framework#37893] [Ideas] [answered]
I use Laravel for a long time and I will always be updated with the latest PHP versions and Laravel versions. 

And there are some things I would like to know if it's too much start some quiet refactor with some specific things for the newest PHP versions that are coming.

For example in models instead:
```
protected $table = "name";
```

Be like:
```
protected string $table = "name";
```

And so on... I tried to do this by myself on my models but some exceptions are thrown since it's a valid parental extended excepted behavior.

What are your thoughts about that?

**Comment by kohenkatz** [comment]
PHP 7.3 is still receiving security updates until the end of 2021 and PHP 7.4 will be supported until the end of 2022. It is probably still too early to force people to use 8.x.

**Reply by tpetry** [reply]
Laravel 9 will depend on Symfony 6, which will depend on PHP 8. So the minimum requirement will be PHP 8.

**Comment by tpetry** [comment, marked as answer]
I asked the same on Twitter and Taylor [answered](https://twitter.com/taylorotwell/status/1412809700962451459?s=21) he wants to limit the breaking changes, which is completely understandable.
